bundle/config/validate/schema_references_test.go

```diff
@@ -0,0 +1,228 @@
+	pipelineTargetL := dyn.Location{File: "file1", Line: 1, Column: 1}
+	pipelineSchemaL := dyn.Location{File: "file2", Line: 2, Column: 2}
+	pipelineL := dyn.Location{File: "file3", Line: 3, Column: 3}
+	schemaL := dyn.Location{File: "file4", Line: 4, Column: 4}
+
+	for _, tc := range []struct {
+		schemaV  string
+		targetV  string
+		catalogV string
+		want     diag.Diagnostics
+	}{
+		{
+			schemaV:  "",
+			targetV:  "",
+			catalogV: "",
+			want:     diag.Diagnostics{},
+		},
+		{
+			schemaV:  "",
+			targetV:  "",
+			catalogV: "main",
+			want: diag.Diagnostics{{
+				Summary:  "Unity Catalog pipeline should have a schema or target defined",
+				Severity: diag.Error,
+				Detail: `The target or schema field is required for UC pipelines. Reason: DLT
+requires specifying a target schema for UC pipelines. Please use the
+TEMPORARY keyword in the CREATE MATERIALIZED VIEW or CREATE STREAMING
+TABLE statement if you do not wish to publish your dataset.`,
+				Locations: []dyn.Location{pipelineL},
+				Paths: []dyn.Path{
+					dyn.MustPathFromString("resources.pipelines.p1.schema"),
+					dyn.MustPathFromString("resources.pipelines.p1.target"),
+				},
+			}},
+		},
+		{
+			schemaV:  "both",
+			targetV:  "both",
+			catalogV: "main",
+			want: diag.Diagnostics{{
+				Severity:  diag.Error,
+				Summary:   "Both schema and target are defined in a Unity Catalog pipeline. Only one of them should be defined.",
+				Locations: []dyn.Location{pipelineSchemaL, pipelineTargetL},
+				Paths: []dyn.Path{
+					dyn.MustPathFromString("resources.pipelines.p1.schema"),
+					dyn.MustPathFromString("resources.pipelines.p1.target"),
+				},
+			}},
+		},
+		{
+			schemaV:  "schema1",
+			targetV:  "",
+			catalogV: "other",
+			want:     diag.Diagnostics{},
+		},
+		{
+			schemaV:  "schema1",
+			targetV:  "",
+			catalogV: "main",
+			want: diag.Diagnostics{{
+				Severity: diag.Warning,
+				Summary:  `Use ${resources.schemas.s1.name} syntax to refer to the UC schema instead of directly using its name "schema1"`,
+				Detail: `Using ${resources.schemas.s1.name} will allow DABs to capture the deploy time dependency this DLT pipeline
+has on the schema "schema1" and deploy changes to the schema before deploying the pipeline.`,
+				Locations: []dyn.Location{pipelineSchemaL, schemaL},
+				Paths: []dyn.Path{
+					dyn.MustPathFromString("resources.pipelines.p1.schema"),
+					dyn.MustPathFromString("resources.schemas.s1"),
+				},
+			}},
+		},
+		{
+			schemaV:  "",
+			targetV:  "schema1",
+			catalogV: "main",
+			want: diag.Diagnostics{{
+				Severity: diag.Warning,
+				Summary:  `Use ${resources.schemas.s1.name} syntax to refer to the UC schema instead of directly using its name "schema1"`,
+				Detail: `Using ${resources.schemas.s1.name} will allow DABs to capture the deploy time dependency this DLT pipeline
+has on the schema "schema1" and deploy changes to the schema before deploying the pipeline.`,
```

Comment:
It would be painful to update this test if the message changes, but it's not nice to assert on the message.

I have added a "golden files" test utilities here https://github.com/databricks/cli/pull/2025 , it's a good fit for this cases.

Full output is stored in the file and can be updated by running tests with TESTS_OUTPUT=OVERWRITE

Reply:
Note, that PR only targets integration tests for now, so not ready for your use case yet.

Reply:
I think it's fine in this case since its unit tests that are colocated with the mutator that implements the functionality. It'll be a simple search and replace when updating.

Are you proposing to extend the golden file approach to mutators or unit tests? Possibly by serializing the diagnostics and persisting them in a file?